docs/workflow.md

```diff
@@ -0,0 +1,75 @@
+# Workflow
+
+## Overview
+
+The development workflow is designed to focus on the raw Helm Chart development and then simulate the deployment
+via a local Kubernetes cluster installed with ArgoCD. This allows developers to iterate quickly on their plugins
+locally without needing to deploy to a production environment until they are ready. The workflow is as follows:
+
+1. Create the plugin locally and push it to a remote branch.
+2. Launch a local Kind Cluster pre-installed with ArgoCD.
+3. Install the ArgoCD Application Resource that points to your plugin on your remote branch.
+4. Log in to the ArgoCD UI and see your plugin listed as an Application.
+5. Make changes to your plugin locally and push them to the remote branch.
+6. ArgoCD will automatically detect the changes and update the application in the local cluster.
```

Comment:
we should probably mention doing a make package-PLUGIN I see you mention this later int he Advanced section, but probably should mention it here as well.

Reply:
Added